orchard-ws/app/services/Metric.scala

```diff
@@ -0,0 +1,99 @@
+package services
+
+import java.io.StringWriter
+import javax.inject.{Inject, Singleton}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import com.typesafe.config.ConfigFactory
+import play.api.Configuration
+import play.api.mvc.RequestHeader
+import io.prometheus.client._
+import io.prometheus.client.exporter.common.TextFormat
+import io.prometheus.client.hotspot.DefaultExports
+
+trait Metric {
+  private lazy val config = new Configuration(ConfigFactory.load())
+
+  private val enableMetrics =
+    config.getOptional[Boolean]("orchard.metrics.enable").getOrElse(false)
+
+  private val bypassPaths = config.getOptional[Seq[String]]("orchard.metrics.bypass.paths") match {
+    case Some(paths) => paths.toSet
+    case _ => Set[String]()
+  }
+
+  private val staticPathMarkers = Seq("instance", "__metrics", "__status")
+
+  def parseRequest(request: RequestHeader): Option[(String, String)] = {
+    if (checkPathIsDisabled(request.path) || !checkMetricIsEnabled)
+      None
+    else {
+      val (staticPath, args) = request.path
+        .split("/")
+        .filter(_.nonEmpty)
+        .foldLeft((List[String](), List[String]())) {
+          case ((Nil, l2), token) => (List(token), l2)
+          case ((h :: t, l2), token) =>
+            if (staticPathMarkers.contains(h)) (h :: t, token :: l2)
+            else (token :: h :: t, l2)
+        }
+      Some((staticPath.reverse.mkString("-"), args.reverse.mkString("-")))
+    }
+  }
+
+  def checkMetricIsEnabled: Boolean = { enableMetrics }
+
+  def checkPathIsDisabled(path: String): Boolean = { bypassPaths.contains(path) }
+
+  def incrementStatusCount(status: String): Unit
+
+  def startApiTimer(labels: String*): () => Unit
+
+  def collect: Future[String]
+
+  def clear(): Unit
+
+}
+
+@Singleton
+class PrometheusMetric @Inject() (implicit ec: ExecutionContext) extends Metric {
```

Comment:
@Terkwood good that you worked out the test error.  

Another option to deal with that is to move the metric register to a companion `object PrometheusMetric`.  The Play Singleton did not work as well as the Scala `object`

Reply:
Clarified offline with Simon that we can follow this path in a future refactor to use kineticpulse -- marking resolved